lib/ecto/changeset/relation.ex

```diff
@@ -175,20 +175,16 @@ defmodule Ecto.Changeset.Relation do
   def on_replace(%{on_replace: :raise, field: name, owner: owner}, _) do
     raise """
     you are attempting to change relation #{inspect name} of
-    #{inspect owner}, but there is missing data.
+    #{inspect owner}, but the `:on_replace` option of this
+    relation is set to `:raise`.
 
-    If you are attempting to update an existing entry, please make sure
-    you include the entry primary key (ID) alongside the data.
+    By default, if you don't specify it, Ecto sets this option to
+    `:raise`, therefore not allowing you to change relations or
+    embeds through the parent record's changeset.
```

Comment:
This is not true. 
The error is only raised if data is missing, i.e. some of the objects that are currently associated were not accounted for in the params. It's not raised on every change.